client/requestUtil.js

```diff
@@ -420,6 +429,49 @@ RequestUtil.prototype.put = function (category, record) {
   })
 }
 
+/**
+ * Compact all records in a category
+ * @param {string=} category - the category ID
+ */
+RequestUtil.prototype.compactCategory = function (category) {
+  const thisCategory = category
+  if (!recordUtil.CATEGORY_IDS.includes(thisCategory)) {
+    throw new Error(`Unsupported sync category: ${category}`)
+  }
+
+  const prefix = `${this.apiVersion}/${this.userId}/${category}`
+  let options = {
+    MaxKeys: 1000,
+    Bucket: this.bucket,
+    Prefix: prefix
+  }
+  this.withRetry(() => {
+    return s3Helper.listObjects(this.s3, options, false)
+  }).then((s3Objects) => {
+    // This is used to keep the most recent records for each object id
+    let latestRecords = {}
+    let s3ObjectsToDelete = []
+    const recordObjects = this.s3ObjectsToRecords(s3Objects.contents)
+    recordObjects.forEach((recordObject) => {
+      const record = recordObject.record
+      if (latestRecords[record.objectId]) {
+        console.log('compaction deletes')
+        if (record.syncTimestamp > latestRecords[record.objectId].record.syncTimestamp) {
+          s3ObjectsToDelete = s3ObjectsToDelete.concat(latestRecords[record.objectId].objects)
+          console.log(latestRecords[record.objectId].record)
```

Comment:
@darkdh  can you use here `${JSON.stringify(latestRecords[record.objectId].record)}`?
right now it looks on bash console as 
```
[12644:12644:1022/150032.455150:INFO:CONSOLE(1)] "compaction deletes", source: chrome-extension://nomlkjnggnifocmealianaaiobmebgil/extension/brave-sync/bundles/bundle.js (1)
[12644:12644:1022/150032.455640:INFO:CONSOLE(1)] "[object Object]", source: chrome-extension://nomlkjnggnifocmealianaaiobmebgil/extension/brave-sync/bundles/bundle.js (1)
```
and in chromium inspector console it requires to expand and the text search cannot find hidden strings?

Reply:
that is intentional for inspector console to see better structure rather than a single line of string. We should always check inspector console which can be seen for debug and non debug build of brave-core. For the search, the entry will be highlighted, all we need to do is expand it.